### ISSUE TITLE
How to handle cancelling a run from CLI without page crashing?

### ISSUE THREAD

**Issue by chrisheseltine** [puppeteer/puppeteer#5735]
What I am aiming to do is super simple, and I'm sure it used to work.

I have built a basic web scraper using Puppeteer with JS, and when I am doing a run, say for testing, I want to cancel that run once I see my new feature is working.

When I do this I get a tonne of nasty error spam about how the page crashed and I didn't handle it.

I have setup listeners on the page object for "error" and "pageerror" events as per the docs but I am still getting this issue. I have also wrapped the entire application in a try/catch.

Does anyone know how to prevent this? Say for example on event "cancel"...

`(node:7908) UnhandledPromiseRejectionWarning: Error: Page crashed!
    at Page._onTargetCrashed (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\puppeteer\lib\Page.js:215:24)
    at CDPSession.<anonymous> (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\puppeteer\lib\Page.js:123:56)
    at CDPSession.emit (events.js:311:20)
    at CDPSession._onMessage (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\puppeteer\lib\Connection.js:200:12)
    at Connection._onMessage (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\puppeteer\lib\Connection.js:112:17)
    at WebSocket.<anonymous> (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\puppeteer\lib\WebSocketTransport.js:44:24)
    at WebSocket.onMessage (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\ws\lib\event-target.js:120:16)
    at WebSocket.emit (events.js:311:20)
    at Receiver.receiverOnMessage (C:\Users\hesel\dev\repo\theseedlist\scrapers\seedsman-com\node_modules\ws\lib\websocket.js:789:20)
    at Receiver.emit (events.js:311:20)
(node:7908) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:7908) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
$ ERROR: The process with PID 2752 (child process of PID 7908) could not be terminated.
Reason: There is no running instance of the task.
`

**Comment by munrocket** [comment]
This problem is [was appear](https://github.com/munrocket/puppeteer-webgl-ci/pull/2) in puppeter@v3.0.0 and I suggest to downgrade version.
```
(node:5227) UnhandledPromiseRejectionWarning: Error: Page crashed!
    at Page._onTargetCrashed (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/puppeteer/lib/Page.js:201:28)
    at CDPSession.Page.client.on.event (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/puppeteer/lib/Page.js:121:60)
    at CDPSession.emit (events.js:198:13)
    at CDPSession._onMessage (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/puppeteer/lib/Connection.js:168:18)
    at Connection._onMessage (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/puppeteer/lib/Connection.js:88:25)
    at WebSocketTransport._ws.addEventListener.event (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/puppeteer/lib/WebSocketTransport.js:24:32)
    at WebSocket.onMessage (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/ws/lib/event-target.js:120:16)
    at WebSocket.emit (events.js:198:13)
    at Receiver.receiverOnMessage (/home/travis/build/munrocket/puppeteer-webgl-ci/node_modules/ws/lib/websocket.js:800:20)
    at Receiver.emit (events.js:198:13)
(node:5227) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:5227) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Steps to reproduce**

```
git clone https://github.com/munrocket/puppeteer-webgl-ci
cd puppeteer-webgl-ci
npm un puppeteer && npm i -D puppeteer@3.0.1 && npm install
CI=2 npm run e2e
```

**Comment by munrocket** [comment]
My problem solved 15 min ago by yourself (merged by bot) with latest node and puppeteer. O.o

**Comment by timvandermeij** [comment]
I'm having this problem as well while introducing Puppeteer for Mozilla's PDF.js project in https://github.com/mozilla/pdf.js/pull/11807. It only happens on Windows. I have managed to resolve the "Page crashed!" error by closing all pages explicitly before closing the browser as mentioned in https://github.com/puppeteer/puppeteer/issues/2269#issuecomment-491009218, so only the "process could not be terminated" error remains. The error does indicate that the browser is closed, so I'm not sure if this error is actually important?!

**Comment by munrocket** [comment]
In future we can have a problems, because 2.1.1 only with experimental WebGPU renderer.
It's probably my fault in test script, but all was ok in previous version.

**Comment by munrocket** [comment]
Somehow this problem was fixed in 3.1.0. I am just updated puppeteer in three.js repo and all is ok 😄
